src/fullnode.rs

```diff
@@ -24,30 +24,42 @@ pub struct FullNode {
     pub thread_hdls: Vec<JoinHandle<()>>,
 }
 
+pub enum LedgerFile {
```

Comment:
This should be separate types for input and output. Ideally, both reduce to Option(OsPath).

Reply:
this feels like overtyping
```
pub enum InFile {
    StdIn,
    Path(String),
}

pub enum OutFile {
    StdOut,
    Sink,
    Path(String),
}
```

Reply:
`infile: InFile, outfile: Option<OutFile>` so not so bad, +1 more then combined.